dist/bin/current/helperScripts/setupRuntimeEnvironment.sh

```diff
@@ -31,14 +31,44 @@ checkAndSetJDKInFolder ${PWD}/dist
 
 [[ -z $JAVA_HOME ]] && checkAndSetJDKInFolder ~/.roddy
 
-[[ -z $JAVA_HOME ]] && echo "There was no java runtime environment or jdk setup. Roddy cannot be compiled." && exit 1
-[[ -z $GROOVY_HOME ]] && echo "Groovy SDK / Runtime not found, Roddy cannot be compiled or started." && exit 1
+if [[ ! -z $JAVA_HOME && ! -z $GROOVY_HOME ]]; then
+  PATH=$JDK_HOME/bin:$JAVA_HOME/bin:$GROOVY_HOME/bin:$PATH
+#  JFX_LIBINFO_FILE=~/.roddy/jfxlibInfo
+#  if [[ ! -f ${JFX_LIBINFO_FILE} ]] || [[ ! -f `cat ${JFX_LIBINFO_FILE}` ]]; then
+#    echo `find ${JAVA_HOME}/ -name "jfxrt.jar"` > ${JFX_LIBINFO_FILE}
+#  fi
+  [[ -z $JAVA_HOME ]] && echo "There was no java runtime environment or jdk setup. Roddy cannot be compiled." && exit 1
+  [[ -z $GROOVY_HOME ]] && echo "Groovy SDK / Runtime not found, Roddy cannot be compiled or started." && exit 1
+else
+
+  # Check, if Java can be called and extract the version
+  # OpenJDK and Sun JDK should both work from version 8 on.
+  javaBinary=`which java 2> /dev/null`
+  javaSearchError=$?
+  groovyBinary=`which groovy 2> /dev/null`
+  groovySearchError=$?
+
+  [[ $javaSearchError == 1 ]]   && echo "There was no java runtime environment or jdk setup. Roddy cannot be compiled." && exit 1
+  [[ $groovySearchError == 1 ]] && echo "Groovy SDK / Runtime not found, Roddy cannot be compiled or started." && exit 1
 
-PATH=$JDK_HOME/bin:$JAVA_HOME/bin:$GROOVY_HOME/bin:$PATH
-JFX_LIBINFO_FILE=~/.roddy/jfxlibInfo
-if [[ ! -f ${JFX_LIBINFO_FILE} ]] || [[ ! -f `cat ${JFX_LIBINFO_FILE}` ]]; then
-	echo `find ${JAVA_HOME}/ -name "jfxrt.jar"` > ${JFX_LIBINFO_FILE}
 fi
 
 >&2 echo `which java`
 >&2 echo `which groovy`
+
+function compareVersion {
```

Comment:
Could you document a bit?
compareVersion $language $actualMajorDotMinor $requiredMajorDotMinor?

Also an early check for != null here can prevent lengthly debugging if something goes wrong in the parsing.